File: java/2-finish/akka-streams/src/main/java/pipelines/workshop/ingress/CardPaymentHttpIngress.java
package pipelines.workshop.ingress;

import akka.http.javadsl.common.EntityStreamingSupport;
import akka.http.javadsl.marshallers.jackson.Jackson;
import pipelines.akkastream.AkkaServerStreamlet;
import pipelines.akkastream.StreamletLogic;
import pipelines.akkastream.util.javadsl.HttpServerLogic;
import pipelines.streamlets.StreamletShape;
import pipelines.streamlets.avro.AvroOutlet;
import pipelines.workshop.schema.java.CardPayment;

public class CardPaymentHttpIngress extends AkkaServerStreamlet {

    AvroOutlet<CardPayment> outlet = AvroOutlet.<CardPayment>create("out", CardPayment.class);

    @Override
    public StreamletShape shape() {
        return StreamletShape.createWithOutlets(outlet);
    }

    @Override
    public StreamletLogic createLogic() {
        EntityStreamingSupport ess = EntityStreamingSupport.json();
        return HttpServerLogic.createDefaultStreaming(this, outlet, Jackson.byteStringUnmarshaller(CardPayment.class), ess, getStreamletContext());
    }
}

File: scala/2-finish/akka-streams/src/main/scala/pipelines/workshop/ingress/CardPaymentHttpIngress.scala
package pipelines.workshop.ingress

import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
import pipelines.akkastream.util.scaladsl.HttpServerLogic
import pipelines.akkastream.{ AkkaServerStreamlet, StreamletLogic }
import pipelines.streamlets.StreamletShape
import pipelines.streamlets.avro.AvroOutlet
import pipelines.workshop.ingress.JsonFormats._
import pipelines.workshop.schema.CardPayment

object CardPaymentHttpIngress extends AkkaServerStreamlet {

  val out = AvroOutlet[CardPayment]("out", payment ⇒ payment.customerId)
  val shape = StreamletShape.withOutlets(out)

  override protected def createLogic(): StreamletLogic =
    HttpServerLogic.default(this, out)
}
